app/config.py

```diff
@@ -20,6 +22,39 @@ def get_abs_path(file_path: str):
         return os.path.join(ROOT_DIR, file_path)
 
 
+def parse_environment(env_var: str) -> Optional[Any]:
+    """Convert environmental string into Python object
+
+    Args:
+        env_var (str): environment constant, example: '["domain1.org", "domain2.com"]'
+
+    Returns:
+        Any
+    """
+    try:
+        return literal_eval(env_var)
+    except Exception as e:
+        print("This expression is failed to parsed: %s, with reason: %s", (env_var, e))
+
+
+def sl_get_env(key: str, default_factory: Optional[Callable] = None) -> Any:
+    """Convert environment key into Python object
+
+    Args:
+        key (str): environment constant, example: SL_DB
+        default_factory (Callable, optional): returns value if environment is None. Notice: if this argument and environment variable are not specified, this function will raised. Defaults to None.
+
+    Returns:
+        Any: Python object
+    Raises:
+        ValueError: when key value is empty and default_factory is not specified
+    """
+    env = os.getenv(key)
+    if not (env or default_factory):
+        raise ValueError("This env should be specified: %s", key)
+    return parse_environment(env) or default_factory()
```

Comment:
I think we can replace `parse_environment` by  `literal_eval` and remove `parse_environment()`, this way the code would be a bit simpler :)

Reply:
I think the problem is that `literal_eval` will raised for incorrect expressions, the `try/catch` block will remove this edge case